Specs/Scene/SceneSpec.js

```diff
@@ -1519,16 +1519,22 @@ describe(
         url: "//terrain/tiles",
       });
 
-      expect(scene.terrainProvider).toBe(globe.terrainProvider);
-
-      scene.globe = undefined;
-      expect(function () {
-        scene.terrainProvider = new CesiumTerrainProvider({
-          url: "//newTerrain/tiles",
+      return scene.terrainProvider.readyPromise
+        .catch(() => {
```

Comment:
I'm not sure I follow why all these `catch` blocks are needed. What error is being thrown? Is it expected, or can we tweak something about the tests such that the `readyPromises` resolve?

Reply:
I should have been explicit in the `catch` here too. This is expected to fail since we're providing invalid URLs to the `CesiumTerrainProvider`. I've updated the spec to return valid data by overriding `Resource` load functions (and resetting them after) as is done in the `CesiumTerrainProviderSpec`.